internal/server/models/organization.go

```diff
@@ -30,8 +30,15 @@ type OrganizationMember struct {
 
 func (OrganizationMember) IsOrganizationMember() {}
 
-func (o *OrganizationMember) SetOrganizationID(id uid.ID) {
+type OrganizationIDSource interface {
+	OrganizationID() uid.ID
+}
+
+func (o *OrganizationMember) SetOrganizationID(source OrganizationIDSource) {
```

Comment:
Passing in the whole interface instead of just a value is a bit unfortunate, but I don't see any way to avoid this. 

We need to completely avoid the call to `OrganizationID` when an org ID is set, so we have to defer looking up the ID. Accepting a function seems like the easiest way to do that.